### PR TITLE
Enable the generation of javadoc and scan for issue

### PR DESCRIPTION
**Pull request by laeubi**
This will require:
- https://github.com/eclipse-platform/eclipse.platform.releng.aggregator/pull/614

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -53,6 +53,21 @@
...
+	<profile>
+		<id>javadoc</id>
+		<build>
+			<plugins>
+				<plugin>
+					<groupId>org.apache.maven.plugins</groupId>
+					<artifactId>maven-javadoc-plugin</artifactId>
+					<configuration>
+						<!-- Special code only for J9 -->
+						<excludePackageNames>org.eclipse.osgi.internal.cds</excludePackageNames>
```
Comment (akurtakov) [inline pom.xml:65]: I think we should exclude *.internal.* as javadoc for internals is not generally required (yet).

Comment (laeubi) [reply pom.xml:65]: It is not strictly required, but internal code is still used, so bad javadoc hurts there too, also internal API is widly used across eclipse plugins, so I think its good to have it for now, if it really creates issues, we can later use more explicit excludes.

**Comment by laeubi** [comment]
The test-failure of the Gh actions is unrelated...

File: pom.xml (hunk 1 above)
Comment (akurtakov) [reply pom.xml:65]: OK. Only thing to consider is that if showing hundreds of warnings very few people (if anyone at all) will try to get a clean build as the battle looks lost but if the number is not that great there are people that try to achieve it. Reiteration with stricter rules can happen after that.

Comment (laeubi) [reply pom.xml:65]: I have now enabled some of the options from the isv build, that should reduce warnings/errors. Beside that I hope that at least no **new** ones are introduced that would already be a benefit.

**Pull request merged**